stable_baselines/common/callbacks.py

```diff
@@ -0,0 +1,226 @@
+import os
+from abc import ABC, abstractmethod
+from typing import Union, List
+
+import gym
+import numpy as np
+
+from stable_baselines.common.base_class import BaseRLModel # pytype: disable=pyi-error
+from stable_baselines.common.vec_env import VecEnv, sync_envs_normalization
+from stable_baselines.common.evaluation import evaluate_policy
+
+
+class BaseCallback(ABC):
+    """
+    Base class for callback.
+
+    :param verbose: (int)
+    """
+    def __init__(self, verbose=0):
+        super(BaseCallback, self).__init__()
+        self.model = None
+        self.training_env = None
+        self.n_calls = 0
+        self.num_timesteps = 0
+        self.verbose = verbose
+
+    def init_callback(self, model: BaseRLModel) -> None:
+        """
+        Initialize the callback by saving references to the
+        RL model and the training environment for convenience.
+        """
+        self.model = model
+        self.training_env = model.get_env()
+
+    def on_training_start(self, locals_: dict, globals_: dict) -> None:
+        pass
+
+    # Should we include that?
+    # def on_rollout_start(self, locals_: dict, globals_: dict) -> None:
```

Comment:
For PPO/SAC-like arguments, this would almost same as `step`, no? Or does `on_step` now do on each env step as suggested by next docstring?

Reply:
it would the same unless we unify everything (up to n_envs steps for a VecEnv) which is what we should do I think

Reply:
Ok, in that case `rollout_start` would be useful. I am not quite of the usefulness of `rollout_end` (after updates?). Another option would be to keep the `after_update` callback style we have now.